### PR TITLE
Chasing wooly quadrapeds again

### PR DESCRIPTION
**Pull request by yaleman**
Cleaning up some TODOs and stuff

Checklist

- [x] This pr contains no AI generated code
- [x] cargo fmt has been run
- [x] cargo clippy has been run
- [x] cargo test has been run and passes

### CODE REVIEW

File: server/lib/src/filter.rs [hunk 1]
```diff
@@ -660,10 +660,10 @@ impl FilterComp {
             FilterComp::Or(filters) => {
                 // If all filters are okay, return Ok(Filter::Or())
                 // If any is invalid, return the error.
-                // TODO: ftweedal says an empty or is a valid filter
+                // @ftweedal says an empty or is a valid filter
                 // in mathematical terms.
                 if filters.is_empty() {
-                    return Err(SchemaError::EmptyFilter);
+                    return Ok(FilterComp::Or(vec![]));
```
Comment (Firstyear) [inline server/lib/src/filter.rs:666]: I think these should stay as errors. Is there a reason to make them valid here?

**Review by Firstyear — CHANGES_REQUESTED** [review]

File: server/lib/src/filter.rs (hunk 1 above)
Comment (yaleman) [reply server/lib/src/filter.rs:666]: 'cause an empty `Or` is an empty `Or` and thus valid and thus "false"

Comment (Firstyear) [reply server/lib/src/filter.rs:666]: But if someone requests an empty or, shouldn't we tell them "hey you're query looks a bit weird?" Is there a use case you are trying to resolve here?

Comment (yaleman) [reply server/lib/src/filter.rs:666]: I dunno, I was mainly just cleaning up TODOs 😄 I can roll it back if you think it's an error.

> TODO: ftweedal says an empty or is a valid filter in mathematical terms.

Comment (Firstyear) [reply server/lib/src/filter.rs:666]: Also gosh damn that's some *ancient* code.

Comment (yaleman) [reply server/lib/src/filter.rs:666]: Done!

**Review by Firstyear — APPROVED** [review]

**Pull request merged**